ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/privacy/methods/eea/EeaGetTransactionReceipt.java

```diff
@@ -92,18 +92,19 @@ public JsonRpcResponse response(final JsonRpcRequest request) {
     PrivateTransaction privateTransaction;
     String privacyGroupId;
     try {
-      ReceiveResponse receiveResponse = getReceiveResponseFromEnclave(transaction, publicKey);
+      final ReceiveResponse receiveResponse = getReceiveResponseFromEnclave(transaction, publicKey);
       LOG.trace("Received transaction information from Enclave");
 
       final BytesValueRLPInput bytesValueRLPInput =
           new BytesValueRLPInput(BytesValues.fromBase64(receiveResponse.getPayload()), false);
 
       privateTransaction = PrivateTransaction.readFrom(bytesValueRLPInput);
       privacyGroupId = receiveResponse.getPrivacyGroupId();
-    } catch (Exception e) {
-      LOG.error("Failed to fetch transaction from Enclave with error " + e.getMessage());
-      return new JsonRpcSuccessResponse(
-          request.getId(), JsonRpcError.PRIVATE_TRANSACTION_RECEIPT_ERROR);
+    } catch (final EnclaveException e) {
+      if (e.getMessage() != null && e.getMessage().equals("EnclavePayloadNotFound")) {
+        return new JsonRpcSuccessResponse(request.getId(), null);
+      }
+      throw e;
```

Comment:
Should this function throw - or should it return a suitable JsonRpcError - i.e. handlers above this won't be able to perform RPC-specific handling of the error.

Reply:
Yes - it will be handled the same way as if `eth_getTransactionReceipt` failed to read from rocks db.